Cocktails9/Cocktails9/RegisterView.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  RegisterView.swift
+//  Cocktails9
+//
+//  Created by Mateja Mitic on 28.11.24..
+//
+
+
+import SwiftUI
+
+struct RegisterView: View {
+    @State private var username: String = ""
+    @State private var email: String = ""
+    @State private var password: String = ""
+    @State private var isLoggedIn: Bool = false
+    @State private var showAlert: Bool = false
+    @State private var alertMessage: String = ""
+    @State private var showRegisterView: Bool = false
+    @State private var currentImageName: String = "bela casa"
+    
+    // Store credentials in UserDefaults
+    private func registerUser() {
+        // Input Validation
+        guard !email.isEmpty, !username.isEmpty, !password.isEmpty else {
+            alertMessage = "All fields must be filled."
+            showAlert = true
+            return
+        }
+        
+        if !isValidEmail(email) {
+            alertMessage = "Please enter a valid email address."
+            showAlert = true
+            return
+        }
+        
+        // Save data to UserDefaults
+        UserDefaults.standard.set(email, forKey: "userEmail")
+        UserDefaults.standard.set(username, forKey: "userUsername")
+        UserDefaults.standard.set(password, forKey: "userPassword")
```

Comment:
This way we won't be able to have multiple users. only one. Can you find a way to name the `forKey` keys in a way to be associated to each registered user? Maybe store a `User` as data using Codable?https://medium.com/@herlinaastari/store-a-custom-object-with-userdefaults-in-swift-5-2bbacfd92c8a